### PR TITLE
chore(ui): debounce effect in timerange query

### PR DESCRIPTION
**Pull request by Marvin9**
Signed-off-by: mayursiinh <marvinduff97@gmail.com>

Fixes https://github.com/prometheus/prometheus/issues/406

### CODE REVIEW

File: web/ui/react-app/src/pages/graph/Panel.tsx [hunk 1]
```diff
@@ -102,6 +105,9 @@ class Panel extends Component<PanelProps, PanelState> {
     this.executeQuery();
   }
 
+  // eslint-disable-next-line
+  debounceExecuteQuery() {}
```
Comment (juliusv) [inline web/ui/react-app/src/pages/graph/Panel.tsx:109]: Could we declare this more like we do for `abortInFlightFetch` above?

**Comment by juliusv** [comment]
Thanks, looks good besides the one comment! Could you rebase this on top of the current main branch? We changed some things in the Netlify build stuff, and that should fix those CI checks.

File: web/ui/react-app/src/pages/graph/Panel.tsx (hunk 1 above)
Comment (Marvin9) [reply web/ui/react-app/src/pages/graph/Panel.tsx:109]: Definitely. It should be like that.

File: web/ui/react-app/src/utils/index.ts [hunk 2]
```diff
@@ -269,3 +269,13 @@ export const parsePrometheusFloat = (value: string): string | number => {
     return Number(value);
   }
 };
+
+export function debounce<Params extends any[]>(func: (...args: Params) => any, timeout: number): (...args: Params) => void {
```
Comment (juliusv) [inline web/ui/react-app/src/utils/index.ts:273]: This is producing some compilation errors on CircleCI due to the restrictive TypeScript settings:

```
Failed to compile.

src/utils/index.ts
  Line 273:41:  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  Line 273:75:  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
```

Comment (juliusv) [reply web/ui/react-app/src/utils/index.ts:273]: (you should also be able to run `make ui-lint` locally)

Comment (Marvin9) [reply web/ui/react-app/src/utils/index.ts]: Done. When I ran `make ui-lint`, it showed `@typescript-eslint/no-explicity-any` warnings at many places. I guess they are written before the CI job was introduced.

**Comment by juliusv** [comment]
:+1: Thanks!

**Pull request merged**